rules/integrations/okta/initial_access_first_occurrence_user_session_started_via_proxy.toml

```diff
@@ -0,0 +1,83 @@
+
+#### Possible investigation steps:
+- Identify the user involved in this action by examining the `okta.actor.id`, `okta.actor.type`, `okta.actor.alternate_id`, and `okta.actor.display_name` fields.
+- Determine the client used by the actor. Review the `okta.client.ip`, `okta.client.user_agent.raw_user_agent`, `okta.client.zone`, `okta.client.device`, and `okta.client.id` fields.
+- Examine the `okta.debug_context.debug_data.flattened` field for more information about the proxy used.
+- Review the `okta.request.ip_chain` field for more information about the geographic location of the proxy.
+- Review the past activities of the actor involved in this action by checking their previous actions.
+- Evaluate the actions that happened just before and after this event in the `okta.event_type` field to help understand the full context of the activity.
+
+### False positive analysis:
+- A user may have legitimately started a session via a proxy for security or privacy reasons.
+
+### Response and remediation:
+- Review the profile of the user involved in this action to determine if proxy usage may be expected.
+- If the user is legitimate and the authentication behavior is not suspicious, no action is required.
+- If the user is legitimate but the authentication behavior is suspicious, consider resetting the user's password and enabling multi-factor authentication (MFA).
+    - If MFA is already enabled, consider resetting MFA for the user.
+- If the user is not legitimate, consider deactivating the user's account.
+- Conduct a review of Okta policies and ensure they are in accordance with security best practices.
+
+## Setup
+The Okta Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.
+"""
+references = [
+    "https://developer.okta.com/docs/reference/api/system-log/",
+    "https://developer.okta.com/docs/reference/api/event-types/",
+    "https://developer.okta.com/docs/reference/api/system-log/#issuer-object",
+    "https://www.elastic.co/security-labs/testing-okta-visibility-and-detection-dorothy",
+    "https://sec.okta.com/articles/2023/08/cross-tenant-impersonation-prevention-and-detection",
+]
+risk_score = 47
+rule_id = "6f1bb4b2-7dc8-11ee-92b2-f661ea17fbcd"
+severity = "medium"
+tags = ["Tactic: Initial Access", "Use Case: Identity and Access Audit", "Data Source: Okta"]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset:okta.system and okta.event_type:user.session.start and okta.security_context.is_proxy:true and not okta.actor.id: okta*
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1133"
+name = "External Remote Services"
+reference = "https://attack.mitre.org/techniques/T1133/"
+
+
+[rule.threat.tactic]
+id = "TA0001"
+name = "Initial Access"
+reference = "https://attack.mitre.org/tactics/TA0001/"
+
+[rule.new_terms]
+field = "new_terms_fields"
+value = ["okta.actor.id"]
```

Comment:
Not sure whether this is possible in Okta, but would it make sense to add a tenant ID there? To make sure that the ID is new to the specific tenant? Just like we do with DRs, by adding for example an agent.id in conjunction with a process.executable.

Reply:
Thanks Ruben, that is a good suggestion! There may be instances where a multi-tenant setup has been done in which `cloud.account.id` is typically used to distinguish between these. If an actor uses Okta to access a shared resource in a multi-tenancy environment with a Proxy for the first time, an alert would occur however if this is normal then alerts would be suppressed unless outside of the window. 

If we add `cloud.account.id` as an additional term, then we isolate "is proxy normal for this user in this tenant", which honestly seems like a good signal. To be honest, I am not aware of an actor would be using multiple tenants? That may in itself be a decent lateral movement detection.

Either way, I'll add to the rule!